### PR TITLE
Split ModerationService into ModerationService and ModerationOperations

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Splits ModerationService into two separate classes: ModerationOperations, which handles command/authorization-required operations, and ModerationService, which handles operations that don't require authorization.

### CODE REVIEW

File: Modix.Services/Moderation/ModerationService.cs [hunk 1]
```diff
@@ -183,11 +152,8 @@ public class ModerationService : IModerationService
         }
```
Comment (JakenVeina) [inline Modix.Services/Moderation/ModerationService.cs]: One of the main goals of splitting here is to decouple auth from any operations that aren't initiated from a frontend. So, `IAuthorizationService` should not be pulled in here. I think the only remaining uses here are to grab `CurrentUserId` and `CurrentGuildId` to record as `createdBy` or `deletedBy` or whatever. For these cases, these values should be passed in as parameters to the service method. So, `IModerationOperations` will just grab `IAuthorizationService.CurrentUserId` and `IAuthorizationService.CurrentGuildId` as needed, while auto-initiated stuff will just grab the `ISelfUser`.

**Comment by jrusbatch** [comment]
@Scott-Caldwell Would you mind rebasing this to get rid of the merge commit?

**Comment by Scott-Caldwell** [comment]
Sure thing. I'm at my parents' house for the next couple of days and won't have access to my computer, so I'll do it when I get back home

**Comment by Scott-Caldwell** [comment]
Closing, because the more I think about this PR, the less I like it. It feels like the better approach would be to add the "user ignores authorization" flag to `IAuthorizationService` and just wrap our auth/rank validations in that. I'm worried that doing this split between services and operations will be confusing and will open us up to tripping up and accidentally using the wrong API.

**Pull request closed without merging**